### PR TITLE
[test](nereids) add filter estimation ut cases

### PR DESCRIPTION
**Pull request by englefly**
# Proposed changes
fix a bug for filter estimation, in pattern of A>10 and A<20.

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/test/java/org/apache/doris/nereids/stats/FilterEstimationTest.java [hunk 1]
```diff
@@ -71,7 +71,9 @@ public void test1() {
         FilterEstimation filterEstimation = new FilterEstimation(stat);
         StatsDeriveResult expected = filterEstimation.estimate(or);
         Assertions.assertTrue(
-                Precision.equals((0.5 * 0.1 + 0.1 - 0.5 * 0.1 * 0.1) * 1000, expected.getRowCount(), 0.01));
+                Precision.equals((0.5 * 0.1 + FilterEstimation.DEFAULT_EQUALITY_COMPARISON_SELECTIVITY
```
Comment (morrySnow) [inline fe/fe-core/src/test/java/org/apache/doris/nereids/stats/FilterEstimationTest.java:74]: plz add comment to explain these magic numbers

File: fe/fe-core/src/test/java/org/apache/doris/nereids/stats/FilterEstimationTest.java [hunk 2]
```diff
@@ -324,4 +326,172 @@ public void test11() {
         StatsDeriveResult estimated = filterEstimation.estimate(not);
         Assertions.assertEquals(1000 * 7.0 / 10.0, estimated.getRowCount());
     }
+
+    //c>100
+    // a is primary-key, a.ndv is reduced
+    // b is normal, b.ndv is not changed
+    // c.selectivity is still 1, but its range becomes half
+    @Test
+    public void test12() {
```
Comment (morrySnow) [inline fe/fe-core/src/test/java/org/apache/doris/nereids/stats/FilterEstimationTest.java:335]: need a meaningful name for each test

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.58 seconds
 load time: 443 seconds
 storage size: 17149814613 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221115125807_clickbench_pr_46529.html

File: fe/fe-core/src/test/java/org/apache/doris/nereids/stats/FilterEstimationTest.java (hunk 1 above)
Comment (englefly) [reply fe/fe-core/src/test/java/org/apache/doris/nereids/stats/FilterEstimationTest.java]: done

File: fe/fe-core/src/test/java/org/apache/doris/nereids/stats/FilterEstimationTest.java (hunk 2 above)
Comment (englefly) [reply fe/fe-core/src/test/java/org/apache/doris/nereids/stats/FilterEstimationTest.java:339]: done

**Review by morrySnow — APPROVED** [review]

**Pull request merged**